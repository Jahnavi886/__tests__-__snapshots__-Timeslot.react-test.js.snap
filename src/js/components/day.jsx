import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import * as moment from 'moment';
import Timeslot from 'timeslot';

import {
  DEFAULT_TIMESLOT_FORMAT,
  DEFAULT_TIMESLOT_SHOW_FORMAT,
  DEFAULT_TIMESLOTS,
} from 'constants/day.js';

export default class Day extends React.Component {

  render() {
    const dayClassNames = classnames({
      'tsc-day': true,
    });

    return (
      <div className={ dayClassNames }>
        { this._renderTimeSlots() }
      </div>
    );
  }

  _renderTimeSlots() {
    const {
      timeslotFormat,
      timeslotShowFormat,
      timeslots,
    } = this.props;

    return timeslots.map((slot, index) => {
      let description = '';
      for (let i = 0; i < slot.length; i ++){
        description += moment(slot[i], timeslotFormat).format(timeslotShowFormat);
        if (i < (slot.length - 1)){
          description += ' - ';
        }
      }
      <Timeslot
        key = { index }
        description = { description }
        onClick = { this._onTimeslotClick.bind(this, index) }
      />;
    });
  }

  _onTimeslotClick(index) {
    const {
      onClick,
    } = this.props;

    onClick(index);
  }
}

Day.defaultProps = {
  timeslotFormat: DEFAULT_TIMESLOT_FORMAT,
  timeslotShowFormat: DEFAULT_TIMESLOT_SHOW_FORMAT,
  timeslots: DEFAULT_TIMESLOTS,
};

/**
 * @type {String} timeslotFormat: format used by moment when identifying the timeslot
 * @type {String} timslotShowFormat: format to show used by moment when formating timeslot hours for final view.
 * @type {Array} timeslots: Array of timeslots.
 * @type {Function} onClick: Function to be excecuted when clicked.
 */
Day.propTypes = {
  timeslotFormat: PropTypes.string.isRequired,
  timeslotShowFormat: PropTypes.string.isRequired,
  timeslots: PropTypes.array.isRequired,
  onClick: PropTypes.func.isRequired,
};
