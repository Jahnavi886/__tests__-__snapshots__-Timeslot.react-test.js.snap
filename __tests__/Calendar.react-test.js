import React from 'react';
import renderer from 'react-test-renderer';
import moment from 'moment';

import Calendar from '../src/js/components/calendar';
import { DEFAULT_TIMESLOTS } from '../src/js/constants/day';

describe('Render tests', () => {
  test('Renders Correctly.', () => {
    const tree = renderer.create(
      <Calendar
        initialDate = { moment().format() }
        timeslots = { DEFAULT_TIMESLOTS }
      />
    )
    .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
