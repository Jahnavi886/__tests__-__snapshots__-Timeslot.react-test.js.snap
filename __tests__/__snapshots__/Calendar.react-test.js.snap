// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render tests Renders Correctly. 1`] = `
<div
  className="tsc-calendar"
>
  <div
    className="tsc-calendar__actions"
  >
    <div
      className="tsc-calendar__action tsc-calendar__action-element tsc-calendar__action-element--left"
      onClick={[Function]}
    >
      ‹
    </div>
    <div
      className="tsc-calendar__action tsc-calendar__action-title"
    >
      June - 2017
    </div>
    <div
      className="tsc-calendar__action tsc-calendar__action-element tsc-calendar__action-element--right"
      onClick={[Function]}
    >
      ›
    </div>
  </div>
  <div
    className="tsc-month"
  >
    <div
      className="tsc-month__actions"
    >
      <div
        className="tsc-month__action tsc-month__action-element tsc-month__action-element--left"
        onClick={[Function]}
      >
        ‹
      </div>
      <div
        className="tsc-month__action tsc-month__action-title"
      >
        4th - 10th
      </div>
      <div
        className="tsc-month__action tsc-month__action-element tsc-month__action-element--right"
        onClick={[Function]}
      >
        ›
      </div>
    </div>
  </div>
</div>
`;
